src/schema/users.ts

```diff
@@ -1176,6 +1193,71 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return streak;
     },
+    streakRecover: async (
+      _,
+      __,
+      ctx: AuthContext,
+    ): Promise<StreakRecoverQueryResult> => {
+      const cantRecoverResult: StreakRecoverQueryResult = {
+        canDo: false,
+        cost: 0,
+        oldStreakLength: 0,
+      };
+
+      const { userId } = ctx;
+      if (!userId) {
+        return cantRecoverResult;
+      }
+
+      const key = generateStorageKey(
+        StorageTopic.Streak,
+        StorageKey.Reset,
+        userId,
+      );
+
+      const oldStreakLength = Number(await getRedisObject(key));
+      const userDontHaveOldStreak =
+        !!oldStreakLength && !isNumber(oldStreakLength);
+      if (userDontHaveOldStreak) {
+        return cantRecoverResult;
+      }
+
+      const streak = await ctx.con.getRepository(UserStreak).findOneBy({
+        userId,
+      });
+
+      if (!streak) {
+        return cantRecoverResult;
+      }
+
+      const { currentStreak } = streak;
+      const streakIsTooLongToRecover = currentStreak > 1;
+      if (streakIsTooLongToRecover) {
+        await ctx.con.getRepository(Alerts).update(
+          {
+            userId,
+          },
+          {
+            showRecoverStreak: false,
+          },
+        );
+        return cantRecoverResult;
+      }
```

Comment:
I am worried having an `update` inside a query, can we update this somewhere else, eg. from the frontend once user dismisses the modal or worker when streak updates to more then 1?

Reply:
Yes, I think that moving to a single mutation would be ok, do we have something similar for other Alerts?

Reply:
we have a generic mutation to update alerts, its called `updateUserAlerts` and used from `AuthContext` `updateAlerts` from frontend.